web/hooks/use-contract-supabase.ts

```diff
@@ -26,6 +31,38 @@ export const usePublicContracts = (contractIds: string[] | undefined) => {
   return contracts
 }
 
+export function useRealtimeContract(contractId: string | undefined) {
+  const [contract, setContract] = useState<Contract | undefined | null>(
+    undefined
+  )
+  useEffectCheckEquality(() => {
+    if (contractId) {
+      getContract(contractId)
+        .then((result) => setContract(result))
+        .catch((e) => console.log(e))
+    }
+  }, [contractId])
+  const changesLength = useRef(0)
+  const changedContract = useRealtimeChanges('UPDATE', 'contracts', {
+    k: 'id',
+    v: contractId ?? '',
+  })
+  console.log(changedContract)
```

Comment:
This doesn't make sense to me, why would you want this function to maintain a list of changes to the contract over time? I don't think you understood the different realtime abstractions I wrote.

- `useRealtimeChannel` -- this is for when you want to get a callback whenever there is an insert/update/delete. It abstracts away the channel management stuff we were having a hard time with. It's useful if you want to do something quirky when you see the changes that isn't just "have a list of the changes" or "update the thing that changed."
- `useRealtimeChanges` -- this is a small wrapper around `useRealtimeChannel` that stores the list of changes in state.
- `useRealtimeRows` -- this is a convenience wrapper around `useRealtimeChanges` that only subscribes to inserts and returns the list of new rows. This tends to be useful for subscribing to ~append-only collections of things.
- `useSubscription` -- this is for when you want to subscribe to a subset of a table and have a live updating snapshot of what you subscribed to, like a Firestore subscription. It handles the "fetch existing and then apply updates to them as they come in" logic. This seems like the easiest thing for you to use here.